### PR TITLE
add docs about checking out the triggering ref

### PR DESCRIPTION
**Pull request by zeke**
See #20

cc @TingluoHuang

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +21,16 @@ steps:
 - run: npm test
 ```
 
+By default, the branch or tag ref that triggered the workflow will be checked out. If you wish to check out a different branch, specify that using `with.ref`:
```
Comment (jasonkarns) [inline README.md]: This isn't quite right. The branch/tag's HEAD will be checked out, which means (by default) the repo is in a Detached HEAD state. That's different than checking out the branch or tag, which is _not_ a detached HEAD.

Comment (zeke) [reply README.md]: Thanks for the clarification. Do you want to open a followup PR to make these docs more clear?

Comment (sfc-gh-zhwang) [reply README.md:24]: how to checkout to the branch without bing detached?

Comment (zeke) [reply README.md:24]: @sfc-gh-zhwang you should be able to fork the repo, clone it, and work off the main branch.

**Pull request merged**